src/tm-threads.c

```diff
@@ -220,6 +239,27 @@ static int TmThreadTimeoutLoop(ThreadVars *tv, TmSlot *s)
 
  */
 
+/**
+ * \brief Exits running pthread with the given exit code
+ *
+ *
+ * With DPDK enabled, the function encapsulates pthread exit with a check for DPDK worker threads.
+ * If the thread has lcore_id attribute set to a different value than LCORE_ID_ANY, it does not
+ * exit because it is identified as a DPDK worker. DPDK worker threads are joined at a single point.
+ * Threads other than DPDK workers exit with pthred_exit.
+ *
+ */
+static void TmThreadExit(ThreadVars *tv, int64_t exit_code)
+{
+#ifdef HAVE_DPDK
+    if (tv->lcore_id == LCORE_ID_ANY) {
```

Comment:
I have disagree with this flow in which all threads are put under DPDK, this is similar to the early work done https://github.com/vipinpv85/DPDK-Suricata_3.0. The same has been modified not to do so. Please rework

Reply:
This function is only my wrapper to end DPDK worker threads only. I added the function in the latest PR. Possibly I can rename the function. 
But do I understand you correctly that you would like to run only worker threads under DPDK? If so, it is already implemented this way.